terraform/source.go

```diff
@@ -195,7 +195,16 @@ func ToSourceUrl(source string, workingDir string) (*url.URL, error) {
 		return nil, errors.WithStackTrace(err)
 	}
 
-	return parseSourceUrl(rawSourceUrlWithGetter)
+	sourceUrl, err := parseSourceUrl(rawSourceUrlWithGetter)
+	if err != nil {
+		return nil, err
+	}
+
+	// explicitly add prefix git:: for the http scheme to avoid further errors
+	if strings.HasPrefix(sourceUrl.Scheme, "http") {
+		sourceUrl.Scheme = "git::" + sourceUrl.Scheme
+	}
+	return sourceUrl, nil
```

Comment:
How do we know that an http URL is always Git? Terraform supports a variety of [source URLs](https://developer.hashicorp.com/terraform/language/modules/sources), including some http URLs that are _not_ Git.

Reply:
Here we add the default _git_ protocol if the user has not specified anything. Otherwise, we will then get an error when we ask go-getter to clone a repository.
I just thought, that we should add `git` it not only to `https://`, but also to `ssh://`

Reply:
without default protocol `getter.Get` returns 
```
 *errors.errorString error downloading 'https://github.com/gruntwork-io/terraform-aws-lambda': bad response code: 404
```

Reply:
Hm, I'm not sure that's right. I think we need to be more picky and do checks based on the supported [module source URLs](https://developer.hashicorp.com/terraform/language/modules/sources):

- If it starts with `http` and the domain is `github.com`, `gitlab.com`, or `bitbucket.org`, or the URL contains `.git`, then I think we can safely prepend `git::`.
- If it starts with `http` and the domain includes `amazonaws.com`, we prepend `s3::`.
- If it starts with `http` and the domain includes `googleapis.com`, we prepend `gcs::`.

Reply:
Ah, understood. It's redone, thank you.